test/dynamo/test_higher_order_ops.py

```diff
@@ -1535,7 +1556,7 @@ def false_fn(x):
                             if node.op == "placeholder"
                         ]
                     )
-                    self.assertEqual(num_placeholders, 3)
+                    self.assertEqual(num_placeholders, arg_count)
```

Comment:
1) Can you show a graph for one of these cases? Maybe using wrap?
2) We're essentially updating the meaning of a "valid" FX subgraph being passed to a HOP. Previously, the definition of a "valid graph" is that it has no free variables and we rely on Dynamo to lift the free variables. Now, the definition is that a "valid graph" contains no free variables and "no free symbols". Do we need to impose more structure on this, like "all the lifted symints are at the end of the function?" (How does Dynamo / AOTAutograd handle this in general?)
3) Similar to the last question: is there something we can do to make this invariant clearer? For example, it would be nice to assert that if the input is an FX graph, then it has these properties. Unfortunately that's expensive, so maybe we should have a debug mode that gets used only in PyTorch tests for this.
4) You probably need to update make_fx at some point to be able to return a graph with symints. Consider the case where we are testing torch.ops.higher_order.cond (and not using torch.cond, which turns on Dynamo, directly). The existing make_fx code will not produce a "valid subgraph" for cond.